core/app-framework/app-native-shared/attr_container.c

```diff
@@ -569,11 +603,14 @@ attr_container_get_attr(const attr_container_t *attr_cont, const char *key)
         val.j = 0;                                                           \
         type = *(uint8_t *)addr++;                                           \
         switch (type) {                                                      \
+            case ATTR_TYPE_BYTE:                                             \
             case ATTR_TYPE_SHORT:                                            \
             case ATTR_TYPE_INT:                                              \
             case ATTR_TYPE_INT64:                                            \
-            case ATTR_TYPE_BYTE:                                             \
+            case ATTR_TYPE_UINT8:                                            \
             case ATTR_TYPE_UINT16:                                           \
+            case ATTR_TYPE_UINT32:                                           \
+            case ATTR_TYPE_UINT64:                                           \
```

Comment:
Had better change the order of these types to be same with the order defined in attr_container.h.

Reply:
And should also add ATTR_TYPE_INT16?

Reply:
Do you mean this？And the order of `switch case` is the same as the order defined in `attr_container.h`. I don't understand your mean of the same order. 
```cpp
/* Attribute type */
enum {
    ATTR_TYPE_BEGIN = 0,
    ATTR_TYPE_BYTE = ATTR_TYPE_BEGIN,
    ATTR_TYPE_INT8 = ATTR_TYPE_BYTE,
    ATTR_TYPE_SHORT,
    ATTR_TYPE_INT16 = ATTR_TYPE_SHORT,
    ATTR_TYPE_INT,
    ATTR_TYPE_INT64,
    ATTR_TYPE_UINT8,
    ATTR_TYPE_UINT16,
    ATTR_TYPE_UINT32,
    ATTR_TYPE_UINT64,
    ATTR_TYPE_FLOAT = 10,
    ATTR_TYPE_DOUBLE,
    ATTR_TYPE_BOOLEAN,
    ATTR_TYPE_STRING,
    ATTR_TYPE_BYTEARRAY,
    ATTR_TYPE_END = ATTR_TYPE_BYTEARRAY
};
```

Reply:
How about:
```C
enum {
    ATTR_TYPE_BEGIN = 0,
    ATTR_TYPE_BYTE = ATTR_TYPE_BEGIN,
    ATTR_TYPE_INT8 = ATTR_TYPE_BYTE,
    ATTR_TYPE_SHORT,
    ATTR_TYPE_INT16 = ATTR_TYPE_SHORT,
    ATTR_TYPE_INT,
    ATTR_TYPE_INT32 = ATTR_TYPE_INT,
    ATTR_TYPE_INT64,
    ATTR_TYPE_UINT8,
    ATTR_TYPE_UINT16,
    ATTR_TYPE_UINT32,
    ATTR_TYPE_UINT64,
    ATTR_TYPE_FLOAT,
    ATTR_TYPE_DOUBLE,
    ATTR_TYPE_BOOLEAN,
    ATTR_TYPE_STRING,
```
And add APIs for get/set:
```
attr_container_get_as_byte
attr_container_get_as_int8
attr_container_get_as_short
attr_container_get_as_int16
attr_container_get_as_int
attr_container_get_as_int32
attr_container_get_as_int64

attr_container_get_as_uint8
attr_container_get_as_uint16
attr_container_get_as_uint32
attr_container_get_as_uint64
```

Reply:
For the order, I mean to handle the cases of switch (code of here and other places) with the same order defined in the `enum`:
```C
        switch (type) {                                                      \
            case ATTR_TYPE_BYTE: /* = ATTR_TYPE_INT8 */                 \
            case ATTR_TYPE_SHORT: /* = ATTR_TYPE_INT16 */           \
            case ATTR_TYPE_INT: /* = ATTR_TYPE_INT32 */                 \
            case ATTR_TYPE_INT64:                                            \
            case ATTR_TYPE_UINT8:                                            \
            case ATTR_TYPE_UINT16:                                           \
            case ATTR_TYPE_UINT32:                                           \
            case ATTR_TYPE_UINT64:
```
Your code looks correct, but had better add comments for `ATTR_TYPE_INT8/INT16/INT32`

Reply:
Ok, thank you very much! I understand what you mean now.